stackstac/reader_protocol.py

```diff
@@ -113,11 +125,16 @@ class FakeReader:
     or inherent to the dask graph.
     """
 
-    def __init__(self, *, dtype: np.dtype, **kwargs) -> None:
+    def __init__(
+        self, *, bands: Optional[Sequence[int]], dtype: np.dtype, **kwargs
+    ) -> None:
         self.dtype = dtype
+        self.ndim = len(bands) if bands is not None else 1
```

Comment:
Oof, `ndim` is not the right name for this, since this is just the length of one dimension... it was a late night.